packages/ai-chat/src/common/chat-session-summary-agent.ts

```diff
@@ -0,0 +1,118 @@
+// *****************************************************************************
+// Copyright (C) 2025 EclipseSource GmbH and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import {
+    Agent,
+    CommunicationRecordingService,
+    CommunicationRequestEntryParam,
+    getTextOfResponse,
+    LanguageModelRegistry,
+    LanguageModelRequirement,
+    PromptService,
+    PromptTemplate,
+    UserRequest
+} from '@theia/ai-core';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { ChatSession } from './chat-service';
+import { generateUuid } from '@theia/core';
+
+const CHAT_SESSION_SUMMARY_PROMPT = {
```

Comment:
Question 6: Should this be an agent? I think it is nice as it gets all the infrastucture. The only confusing thing might be if we allow use case specific prompts for the summarization. We could treat the prompt of this agent as the default though and mention in the doc part that this might be overriden by specific calls.

Reply:
6.: fine

Reply:
This is basically how it is now working: the summary agent is a minimal extension of the abstract agent implementation (useful for e.g. checking history), but it is possible to use other agents.

Reply:
Yeah, I already love it as I can see/edit the prompt and see the history